### DISCUSSION TITLE
Aggregate Relations in a View

### DISCUSSION THREAD

**Discussion by hlanderdev** [pocketbase/pocketbase#2135] [Q&A] [answered]
Is it possible to aggregate single relations into a multi-relation when using `GROUP BY`? It seems the aggregate function that would be used to do this is `JSON_GROUP_ARRAY`, but that does not appear to be supported by the parser. Is there some other way I am missing? Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
All json functions should be supported in the view query. If the SQL expression is a column it must be aliased so that it can yield to valid field name.

Could you share the query you are trying?

**Reply by hlanderdev** [reply]
Ah, yes you are right. I was forgetting to give it an alias. Now it works, but does not behave like a multi-relation and instead a json field. I assume it is an issue similar to what you were talking about in #2029. Here's the query: 

```
SELECT 
    members.id,
    members.email,
    members.first_name,
    members.last_name,
    members.phone_number,
    JSON_GROUP_ARRAY(organization_members.organization_id) as organization_ids
FROM members
LEFT JOIN organization_members ON members.id = organization_members.member_id
GROUP BY members.id
```

**Reply by ganigeorgiev** [reply]
Expressions for now are always detected as `json` because otherwise it will require implementing a full blown SQLite parser to properly understand what columns takes part in the expression and what the expression actually do (even SQLite sometimes doesn't bother parsing expressions; see for example the [`index_xinfo` pragma](https://www.sqlite.org/pragma.html#pragma_index_xinfo) column detection rules).

**Reply by hlanderdev** [reply]
Oh okay, good to know. Thanks for the quick response!
